cache/src/main/java/io/envoyproxy/controlplane/cache/Watch.java

```diff
@@ -1,27 +1,28 @@
 package io.envoyproxy.controlplane.cache;
 
 import io.envoyproxy.envoy.api.v2.DiscoveryRequest;
-import java.util.concurrent.atomic.AtomicBoolean;
+
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
 import java.util.function.Consumer;
 
 /**
  * {@code Watch} is a dedicated stream of configuration resources produced by the configuration cache and consumed by
  * the xDS server.
  */
 public class Watch {
-
+  private static final AtomicIntegerFieldUpdater<Watch> isCancelledUpdater =
+      AtomicIntegerFieldUpdater.newUpdater(Watch.class, "isCancelled");
   private final boolean ads;
-  private final AtomicBoolean isCancelled = new AtomicBoolean();
   private final DiscoveryRequest request;
   private final Consumer<Response> responseConsumer;
-
+  private volatile int isCancelled = 0;
```

Comment:
Why change this? Seems like it's only ever 0 or 1?

Reply:
There is no AtomicBooleanFieldUpdater, it has been suggested to use AtomicIntegerFieldUpdater.

From http://normanmaurer.me/blog/2013/10/28/Lesser-known-concurrent-classes-Part-1/
> On thing to note is that there is no AtomicBooleanFieldUpdater that you could use to replace a AtomicBoolean. This is not a problem, just use AtomicIntegerFieldUpdater with value 0 as false and 1 as true.